docs/index.asciidoc

```diff
@@ -25,21 +25,62 @@ Stream events from files, normally by tailing them in a manner
 similar to `tail -0F` but optionally reading them from the
 beginning.
 
-By default, each event is assumed to be one line and a line is
-taken to be the text before a newline character.
 Normally, logging will add a newline to the end of each line written.
+By default, each event is assumed to be one line
+and a line is taken to be the text before a newline character.
 If you would like to join multiple log lines into one event,
-you'll want to use the multiline codec or filter.
-
-The plugin aims to track changing files and emit new content as it's
-appended to each file. It's not well-suited for reading a file from
-beginning to end and storing all of it in a single event (not even
-with the multiline codec or filter).
+you'll want to use the multiline codec.
+The plugin loops between discovering new files and processing
+each discovered file. Discovered files have a lifecycle, they start off
+in the "watched" or "ignored" state. Other states in the lifecycle are:
+"active", "closed" and "unwatched"
+
+By default, a window of 4095 files is used to limit the number of file handles in use.
+The processing phase has a number of stages:
+
+. checks whether "closed" or "ignored" files have
+changed in size since last time and if so puts them in the "watched" state.
+. selects enough "watched" files to fill the available space in the window,
+these files are made "active".
+. the active files are opened and read, each file is read from the
+last known position to the end of current content (EOF) by default.
+
```

Comment:
Consider an unordered (bulleted) list. (The ordering is implicit in your intro statement.)  Numbered steps usually indicate steps a user needs to follow.  Also consider initial caps for the list items.